### DISCUSSION TITLE
One to One and One to Many same field

### DISCUSSION THREAD

**Discussion by bunyaminduduk** [prisma/prisma#19702] [Q&A]
Hello,

I want to achieve the following:
```prisma
model Event {
  active_meeting Meeting? @relation(name: "active_meeting")

  old_meetings Meeting[] @relation(name: "old_meetings") // I have to put a name, otherwise Prisma will give an error that I need separate names
}

model Meeting {
  event Event? @relation(name: <what name to put here? I want to access the same event of "active_meeting" and "old_meetings">)
}
```

The `Event` model keeps track of the current active meeting and if the active meeting is not active anymore, it goes into `old_meetings` but I don't know what kind of relation I have to put.

I have tried adding an `event_id` but then it is null when the meeting is active until I put it in the `old_meetings` and I don't want that.

Basically: An event can have an active meeting and if it is not active anymore, it will be added to the `old_meetings`. A meeting can only have 1 event which it can access.

**Comment by nurul3101** [comment]
Hello @PhelixTaken 👋 

Can you please try out this schema?

```prisma
model Event {
  id              Int       @id @default(autoincrement())
  activeMeeting   Meeting?  @relation("ActiveMeeting", fields: [activeMeetingId], references: [id])
  activeMeetingId Int?      @unique
  oldMeetings     Meeting[] @relation("OldMeetings")
}

model Meeting {
  id          Int     @id @default(autoincrement())
  eventId     Int?
  activeEvent Event?  @relation("ActiveMeeting")
  oldEvents   Event[] @relation("OldMeetings")
}
```

This should allow you to have a one-to-one and one-to-many relations between `Event` and `Meeting` models.

**Reply by nurul3101** [reply]
I am checking to see if you had a chance to look at my last response.

It would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 
Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Reply by nurul3101** [reply]
@PhelixTaken Do let us know in case you still need any assistance.

**Reply by nurul3101** [reply]
We are closing this Discussion due to inactivity. Please open a new Discussion if you have a similar problem 🙏
